components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -85,19 +87,159 @@ void SolanaTxManager::AddUnapprovedTransaction(
     return;
   }
 
-  SolanaTxMeta meta(from, std::move(tx));
-  meta.set_id(TxMeta::GenerateMetaID());
-  meta.set_origin(
+  auto meta = std::make_unique<SolanaTxMeta>(from, std::move(tx));
+  meta->set_id(TxMeta::GenerateMetaID());
+  meta->set_origin(
       origin.value_or(url::Origin::Create(GURL("chrome://wallet"))));
-  meta.set_created_time(base::Time::Now());
-  meta.set_status(mojom::TransactionStatus::Unapproved);
-  meta.set_chain_id(chain_id);
-  if (!tx_state_manager_->AddOrUpdateTx(meta)) {
+  meta->set_created_time(base::Time::Now());
+  meta->set_status(mojom::TransactionStatus::Unapproved);
+  meta->set_chain_id(chain_id);
+
+  // If the transaction is partially signed, we can't modify the instructions
+  // to add a priority fee.
+  if (meta->tx()->IsPartialSigned()) {
```

Comment:
We should skip the process of adding priority fee instead of rejecting these transactions which are valid.

Reply:
Also for error encounters during the process of adding priority fee, it's probably better to skip adding them too and have only base fee rather than fail the entire op. wdyt?

Reply:
> We should skip the process of adding priority fee instead of rejecting these transactions which are valid.
I think that's what we're doing here. If it's partially signed - then it skips the priority fee stuff and does the existing behavior.

> Also for error encounters during the process of adding priority fee, it's probably better to skip adding them too and have only base fee rather than fail the entire op. wdyt?
That sounds good

Reply:
oh I must misread, we do keep the existing behavior here when partial signed.
Just to double check, do we have an existing test case for partial signed tx?
Thanks.

Reply:
Yes in SolanaTxManagerUnitTest.AddAndApproveTransaction we test that first

Reply:
I updated such that
* If the base fee fetching fails, we still add the transaction and return success but we do not add the gas estimate to the tx meta, and we also don't add any priority fee instructions. This is similar to previous logic and the current logic for for partial signing
* If only base fee fetching succeeds (priority fee fails), we add the estimate to the tx data, but don't add the priority fee instruction
* If base fee succeeds and compute unit fetching succeeds, we add the estimate to the tx data, and we add the priority fee. If the get recent fees rpc call fails, we use the default.

Reply:
Corresponding tests should be added for https://github.com/brave/brave-core/pull/23214/commits/6b78f7c5a954ae36e8ea3eb27a8d190023131aab to make sure every behavior is as expected.